src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,47 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def alert_rule_data_requires_async_lookup(organization, user, data, write_input_channel_id=False):
```

Comment:
@iProgramStuff this function is a little funky in that we expect it to return a boolean value but at the same time it's actually mutating our inputs. Is it possible we could break this up into two functions maybe? One that returns a boolean and another that mutates `action`?

Reply:
Yeah I want to do this better but without duplicating the loop :(

Reply:
I think I could make it short circuit with `return True` if a flag is false and then return a map of slack id to channel name for all actions if a flag is True. That seems cleaner, then it doesn't need to do any weird mutation.

Reply:
@iProgramStuff making a map sounds smart to me

Reply:
I did this. This also cuts down on the slack requests for actions with duplicate channel names (even across multiple triggers).